ipcore/hw_tcl_generator.py

```diff
@@ -0,0 +1,268 @@
+        tcl += ("add_fileset_file " + vhdl_top_level_file + " VHDL PATH " +
+                        vhdl_top_level_file + " TOP_LEVEL_FILE\n")
+        tcl += "# end create_file_sets\n\n\n"
+
+        return tcl
+
+
+    def create_module_assignments(self):
+        tcl = "# # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Created in create_module_assignments\n"
+        tcl += "# # # # # # # # # # # # # # # # # # # #\n\n"
+        tcl += "set_module_assignment embeddedsw.dts.compatible " + \
+            f"dev,{self.vendor}-{self.name}\n"
+        tcl += "set_module_assignment embeddedsw.dts.group autogen \n"
+        tcl += "set_module_assignment embeddedsw.dts.vendor " + \
+            self.vendor + "\n"
+
+        tcl += "# End create_module_assignments\n\n\n"
+        return tcl
+
+
+    def create_connection_point_clock(self):
+        tcl = "# # # # # # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Created by create_connection_point_clock\n"
+        tcl += "# # # # # # # # # # # # # # # # # # # # # #\n\n"
+        tcl += "add_interface clock clock end\n"
+        tcl += "set_interface_property clock clockRate " + \
+            str(self.clock_rate) + "\n"
+        tcl += "set_interface_property clock ENABLED true\n"
+        tcl += "set_interface_property clock EXPORT_OF \"\"\n"
+        tcl += "set_interface_property clock PORT_NAME_MAP \"\"\n"
+        tcl += "set_interface_property clock CMSIS_SVD_VARIABLES \"\"\n"
+        tcl += "set_interface_property clock SVD_ADDRESS_GROUP \"\"\n"
+        tcl += "add_interface_port clock clk clk Input 1\n"
+        tcl += "# End create_connection_point_clock\n\n\n"
+        return tcl
+
+
+    def create_connection_point_reset(self):
+        tcl = "# # # # # # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Created by create_connection_point_reset\n"
+        tcl += "# # # # # # # # # # # # # # # # # # # # # #\n\n"
+        tcl += "add_interface reset reset end\n"
+        tcl += "set_interface_property reset associatedClock clock\n"
+        tcl += "set_interface_property reset synchronousEdges DEASSERT\n"
+        tcl += "set_interface_property reset ENABLED true\n"
+        tcl += "set_interface_property reset EXPORT_OF true\n"
+        tcl += "set_interface_property reset PORT_NAME_MAP \"\"\n"
+        tcl += "set_interface_property reset CMSIS_SVD_VARIABLES \"\"\n"
+        tcl += "set_interface_property reset SVD_ADDRESS_GROUP \"\"\n"
+        tcl += "add_interface_port reset reset reset Input 1\n"
+        tcl += "# End create_connection_point_reset\n\n\n"
+        return tcl
+
+
+    def create_mm_connection_point(self):
+        if not self.has_avalon_mm_slave_signal:
+            return ""
+        tcl = "# # # # # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Created by create_mm_connection_point\n"
+        tcl += "# # # # # # # # # # # # # # # # # # # # #\n\n"
+        memory_slave = 'avalon_slave'
```

Comment:
Any reason to keep this variable around instead of hardcoding it into the generated strings? We don't currently support naming the interfaces anything besides our hardcoded default.

Reply:
Yeah, because even if the user can't rename it on the fly if we decide we want to rename it for the 'Autogen interface' or allow renaming its easier to have a variable there instead of hardcoding it a bunch. Its a bit like the problem of `fe` being hardcoded into a bunch of stuff making renaming everything to use `al` take much longer than if it had been done with a constant